### PR TITLE
Add example that associates data with each point in history

### PR DESCRIPTION
**Pull request by Boscop**
This PR adds an example that associates data with each point in history, as a step towards https://github.com/saschagrunert/yew-router/issues/1.
Also, I noticed that the state wasn't correctly returned by the router, which was probably not noticed previously because only `()` was used for the state.
I fixed it by using [`history.state`](https://developer.mozilla.org/en-US/docs/Web/API/History/state).
Anticipating https://github.com/koute/stdweb/issues/371.
I also removed the unnecessary payload from `Message::BrowserNavigationRouteChanged` since the handler will fetch the route using `Route::current_route()` anyway, and removed the call to `eprintln` which had no effect.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -22,3 +22,8 @@ failure = "0.1.5"
 serde = { version = "1.0.101", features = ["derive"] }
 stdweb = "0.4.18"
 yew = "0.8.0"
+
+[dev-dependencies]
+smart-default = "0.5.2"
+log = "0.4"
```
Comment (saschagrunert) [inline Cargo.toml]: Please use the full version here since dependabot should pick them up and apply updates. :)

File: examples/history_state.rs [hunk 2]
```diff
@@ -0,0 +1,210 @@
+#![recursion_limit = "256"]
+
+#[macro_use]
+extern crate smart_default;
+extern crate yew_router;
+#[macro_use]
+extern crate log;
+extern crate web_logger;
```
Comment (saschagrunert) [inline examples/history_state.rs]: Can we please stick to Rust 2018 and remove the `extern crate`s?

File: Cargo.toml (hunk 1 above)
Comment (Boscop) [reply Cargo.toml]: Done

File: examples/history_state.rs (hunk 2 above)
Comment (Boscop) [reply examples/history_state.rs]: Of course :)
Done

**Comment by saschagrunert** [comment]
Thanks again!

**Pull request merged**